### PR TITLE
packages and vanilla app

### PR DESCRIPTION
**Pull request by hrgdavor**
I have added vanilla app that has threejs and cube #4 

The cube control code we found is unfinished and will require tinkering to sync rotation of cube and main screen and vice-versa

vanilla app work if you run vite in the folder

### CODE REVIEW

File: apps/vanilla-three/src/main.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
+import * as THREE from '@jscadui/bundle-three'
+import { RenderThree } from '@jscadui/render-three'
+import { makeAxes, makeGrid } from '@jscadui/scene'
+import { light } from '@jscadui/themes'
```
Comment (kevinmamaqi) [inline apps/vanilla-three/src/main.js:4]: `import { light as theme } from '@jscadui/themes'` if necessary to use theme naming I rather give it in here.

File: apps/vanilla-three/index.html [hunk 2]
```diff
@@ -0,0 +1,34 @@
+<!DOCTYPE html>
+<html lang="en">
+  <head>
+    <meta charset="UTF-8" />
+    <link rel="icon" type="image/svg+xml" href="/vite.svg" />
+    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+    <title>Vanilla JS + three.js</title>
+    <style>
```
Comment (kevinmamaqi) [inline apps/vanilla-three/index.html:8]: Remove style from here. Add it on an .css file that it's imported in main.jsx if necessary.

File: apps/vanilla-three/src/main.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+window.THREE = THREE // just so we can play with and inspect THREE in browser console
+let theme = light
```
Comment (kevinmamaqi) [inline apps/vanilla-three/src/main.js:7]: Remove theme, use light, or remove theme and use `{ light as theme }` above.

Destrcture the props: `const { grid1, bg, grid2 } = light`

File: packages/render-three/src/CSG2Three.js [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+    const isInstanced = obj.type === 'instances'
+    if ((color || colors) && !isInstanced) {
+      const c = color || colors
+      const opts = {
+        vertexColors: !!colors,
+        opacity: c[3] === undefined ? 1 : c[3],
+        transparent: (color && c[3] !== 1 && c[3] !== undefined) || isTransparent
+      }
+      if (opacity) opts.opacity = opacity
+      if (!colors) opts.color = _CSG2Three.makeColor(color)
```
Comment (kevinmamaqi) [inline packages/render-three/src/CSG2Three.js:46]: Set it as default value

File: packages/render-three/src/CSG2Three.js [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+    let material = materialDef.def
+    const isInstanced = obj.type === 'instances'
+    if ((color || colors) && !isInstanced) {
+      const c = color || colors
+      const opts = {
+        vertexColors: !!colors,
+        opacity: c[3] === undefined ? 1 : c[3],
+        transparent: (color && c[3] !== 1 && c[3] !== undefined) || isTransparent
+      }
+      if (opacity) opts.opacity = opacity
```
Comment (kevinmamaqi) [inline packages/render-three/src/CSG2Three.js:45]: Wouldn't be better an undefined or just set a default value for opacity as well?

**Review by kevinmamaqi — COMMENTED** [review]
Nice, I will play with it ASAP

**Comment by kevinmamaqi** [comment]
I will link the PRs to issues, and try to keep discussions there. Then the naming can be: https://deepsource.io/blog/git-branch-naming-conventions/ - I believe it would be easier to maintain, but also good to get used to it or other workflow asap.

File: packages/render-three/src/CSG2Three.js (hunk 5 above)
Comment (hrgdavor) [reply packages/render-three/src/CSG2Three.js:45]: I think here is more issue of handling user input, I need to check if array is rgb (c[3] === undefined) or it is rgba but opacity is full c[3] === 1

this handling of opacity in general for the 3 engines is something to look at at later point during imrpovement of code.
I would leave it as is for this PR

File: apps/vanilla-three/index.html (hunk 2 above)
Comment (hrgdavor) [reply apps/vanilla-three/index.html:8]: I will add css ref directly in html, I would like to make plain stuff as less depenedent on tooling as possible.

File: apps/vanilla-three/src/main.js (hunk 3 above)
Comment (hrgdavor) [reply apps/vanilla-three/src/main.js:7]: k

File: packages/render-three/src/CSG2Three.js (hunk 4 above)
Comment (hrgdavor) [reply packages/render-three/src/CSG2Three.js:46]: not sure what u mean here :)

**Comment by hrgdavor** [comment]
I have resolved most of the stuff, some I skipped for now. My main issue here is to get ViewBox working ... I have some old code to sync cameras in my old jscad threejs2 prototype

File: apps/engine-test/src/orbitControls.js [hunk 6]
```diff
@@ -0,0 +1,235 @@
...
+      this.target = target
+      this.calc = args[0]
+    } else {
+      this.set(position, target, false)
+    }
+  }
+
+  set(position, target, fire = true) {
+    if (target) this.target = target
+    if (position) this.position = position
```
Comment (kevinmamaqi) [inline apps/engine-test/src/orbitControls.js:36]: The conditional shouldn't be necessary as it already has default values.

File: apps/engine-test/src/orbitControls.js [hunk 7]
```diff
@@ -0,0 +1,235 @@
...
+    this.fireChange()
+  }
+
+  zoomBy(amount) {
+    this.calc.len *= 1 + amount
+    this.position = calcCamPos({ ...this.calc, target: this.target })
+    this.fireChange()
+  }
+
+  clone() {
```
Comment (kevinmamaqi) [inline apps/engine-test/src/orbitControls.js:81]: Why a clone fn?

File: apps/engine-test/src/cameras/camera.js [hunk 8]
```diff
@@ -0,0 +1,69 @@
+import vec3 from 'gl-vec3'
+import mat4 from 'gl-mat4'
+
+import * as orthographicCamera from './orthographicCamera.js'
+import * as perspectiveCamera from './perspectiveCamera.js'
+
+const fromOrthographicToPerspective = (orthographicCamera) => {
+  const { near, far, fov, zoom } = orthographicCamera
+  // recompute projection matrix to use perspective camera projection matrix
+  const { viewport } = orthographicCamera
```
Comment (kevinmamaqi) [inline apps/engine-test/src/cameras/camera.js:10]: deconstruct it on line 8

**Review by kevinmamaqi — APPROVED** [review]
Remove output files (.js.map) files from src files.

It is great, still have to play a bit with it to understand it better.

File: apps/engine-test/src/orbitControls.js (hunk 7 above)
Comment (hrgdavor) [reply apps/engine-test/src/orbitControls.js:81]: it is for future use with animations, OrbitCalc is meant as kind of orbit state, and when I want to animate it I will clone it to have a reference point, then use it to produce intermediate state for animation to next location based on animation percent
.

File: apps/engine-test/src/orbitControls.js (hunk 6 above)
Comment (hrgdavor) [reply apps/engine-test/src/orbitControls.js:36]: conditional is for when you want to change only one of them. There is no default value for the param itself for that same reason.

**Pull request merged**